src/library/scala/concurrent/ExecutionContext.scala

```diff
@@ -139,9 +139,30 @@ object ExecutionContext {
    *
    * @return the global `ExecutionContext`
    */
-  def global: ExecutionContextExecutor = Implicits.global.asInstanceOf[ExecutionContextExecutor]
+  final def global: ExecutionContextExecutor = Implicits.global.asInstanceOf[ExecutionContextExecutor]
 
-  object Implicits {
+  /**
+   * This `ExecutionContext` steals execution time from other threads, by having its
+   * `Runnable`s run on the `Thread` which calls `execute` and then yielding back control
+   * to the caller after *all* its `Runnables` have been executed.
+   * Nested invocations of `execute` will be trampolined to prevent uncontrolled stack space growth.
+   *
+   * WARNING: Do *not* call any blocking code in the `Runnable`s submitted to this `ExecutionContext`
+   * as it will prevent progress by other enqueued `Runnable`s and the calling `Thread`.
+   * In order to maximize application responsiveness, it is strongly advised
+   * to only execute logic which will quickly return control to the caller.
```

Comment:
Is it necessary to say 'In order to maximize application responsiveness' and that it's 'strongly advised'? What are your thoughts on shortening the sentence to just 'Only execute logic which will quickly return control to the caller'?

Reply:
@NthPortal  Updated! :)